### PR TITLE
QueryAPI

### PR DESCRIPTION
**Pull request by mateuszmidor**
KNTK-166

### CODE REVIEW

**Review by Fenthick — CHANGES_REQUESTED** [review]

File: kentikapi/internal/api_endpoints/query.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package api_endpoints
+
+// QuerySQL runs SQL command against all configured devices. Returns top X datasets.
+func QuerySQL() string {
+	return "/query/sql"
+}
+
+// QueryURL returns URL which a logged in user can use to directly access this query in Data Explorer in the Kentik Detect portal.
+func QueryURL() string {
```
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply kentikapi/internal/api_endpoints/query.go:9]: QueryURL is the name of the endpoint, so in the code using it reads as:
api_endpoints.QueryURL()
I can use const strings but decided to keep this consistent with other resources in api_endpoints package.

File: kentikapi/internal/api_payloads/query_test.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	"testing"
+	"time"
+
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/api_payloads"
+	"github.com/kentik/community_sdk_golang/kentikapi/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestQueryChartResponsePNGToQueryChartResult(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (mateuszmidor) [reply kentikapi/internal/api_payloads/query_test.go:14]: We can run a polish-up phase in a spare moment, this test works so let's keep it as it is for now.

File: examples/query_example_test.go [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+	queryObject := models.QueryObject{Queries: []models.QueryArrayItem{queryItem}}
+	models.SetOptional(&queryObject.ImageType, models.ImageTypePNG)
+
+	result, err := client.Query.Chart(context.Background(), queryObject)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Returned chart image type: %s\n", result.ImageType)
+	filename := "/tmp/chart.png"
```
[the remark this answers is not in the document]
Comment (Fenthick) [reply examples/query_example_test.go]: Done

File: kentikapi/internal/api_endpoints/query.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package api_endpoints
+
+// QuerySQL runs SQL command against all configured devices. Returns top X datasets.
```
[the remark this answers is not in the document]
Comment (Fenthick) [reply kentikapi/internal/api_endpoints/query.go]: Done

File: kentikapi/internal/api_payloads/query_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Fenthick) [reply kentikapi/internal/api_payloads/query_test.go:14]: Ok

**Review by Fenthick — APPROVED** [review]

**Review by TedatKentik — APPROVED** [review]
Query API

**Pull request merged**